fitting_source/1D_fitting/plot_6_139La_Ext_Czjzek.py

```diff
@@ -0,0 +1,276 @@
+    """Pre-computes the kernel to use when fitting the experimental spectrum
+
+    Arguments:
+        (np.array) cq_range: Numpy array of cq points
+        (np.array) eta_range: Numpy array of eta points
+        (Method) method: mrsimulator Method object used to simulate the spectra
+        (str) isotope: Isotope of the site
+
+    Returns:
+        Lineshape kernel as a numpy array
+    """
+    Cq, eta = np.meshgrid(cq_range, eta_range, indexing="xy")
+
+    spin_systems = [
+        SpinSystem(sites=[Site(isotope=isotope, quadrupolar=dict(Cq=cq * 1e6, eta=e))])
+        for cq, e in zip(Cq.flatten(), eta.flatten())
+    ]
+    sim = Simulator(spin_systems=spin_systems, methods=[method])
+    sim.config.decompose_spectrum = "spin_system"
+    sim.run(pack_as_csdm=False)
+
+    amp = sim.methods[0].simulation.real
+    return amp
+
+
+# Create ranges to construct cq and eta grid points
+cq_range = np.linspace(0, 100, num=100) * 0.8 + 25
+eta_range = np.arange(21) / 20
+
+kernel = make_kernel(cq_range, eta_range, method, "139La")
+
+
+# %%
+# Make spectrum from kernel and model values
+# ------------------------------------------
+#
+# Next define a function which takes a LMFIT Parameters object, the pre-computed kernel
+# and a signal processor object and returns a CSDM object holding the guess spectrum.
+# Here the Parameters object holds attributes from the
+# :py:class:`~mrsimulator.models.ExtCzjzekDistribution` class which is used to
+# create the probability density function, :math:`{\bf f}`. Then :math:`{\bf f}` is
+# multiplied with the kernel, :math:`{\bf K}`, to produce a spectrum, :math:`{\bf s}`.
+#
+# .. math::
+#
+#     {\bf s} = {\bf K \cdot f},
+#
+# A constant isotropic chemical shift, whose value is also in the Parameters object,
+# is applied to :math:`{\bf s}` using the `Fouier shift relation
+# <https://en.wikipedia.org/wiki/Fourier_transform#Translation_/_time_shifting>`__.
+# Finally, the spectrum is scaled and returned.
+def make_spectrum_from_parameters(
+    params, kernel, processor, cq_range=cq_range, eta_range=eta_range
+):
+    """Makes a spectrum with values given in a parameters object and a pre-computed
+    kernel.
+
+    Arguments:
+        (Parameters) params: LMFIT Parameters object with values
+        (np.array) kernel: Pre-computed spectrum kernel
+        (sp.SignalProcessor) processor: SignalProcessor to apply to spectrum
+
+    Returns:
+        CSDM object of spectrum
+    """
+    # Extract values from parameters object
+    values = params.valuesdict()
+    Cq = values["dist_Cq"]
+    eta = values["dist_eta"]
+    eps = values["dist_eps"]
+    iso_shift = values["dist_iso_shift"]
+
+    # Setup model object and get the amplitude
+    model_quad = ExtCzjzekDistribution({"Cq": Cq, "eta": eta}, eps)
+    _, _, amp = model_quad.pdf(pos=[cq_range, eta_range])
+
+    # Create spectra by dotting the amplitude distribution with the kernel
+    dist = np.dot(amp.flatten(), kernel)
```

Comment:
Isn't the equation `y = K.x`, where `K` is a (*m x n*) kernel, `x` is a column vector of amplitudes of size *n*. `y` is a column vector of spectrum of size *m*?

Reply:
Yes. From briefly reading numpy documentation, a better approach would be 

```kernel.dot(amp.flatten())```

Reply:
Axes of the kernel were swapped. Adjusted the kernel function to return the kernel with the correct shape and now the dot works in the correct order